plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +49,15 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
+	for _, predicates := range f.FailedPredicates {
 		for _, pred := range predicates {
-			reasons = append(reasons, pred.GetReason())
+			reasons.Incr(pred.GetReason())
 		}
-		reasonMsg := fmt.Sprintf("fit failure on node (%s): %s\n", node, strings.Join(reasons, ", "))
-		buf.WriteString(reasonMsg)
 	}
+	reasonMsg := fmt.Sprintf("fit failure summary on nodes : %v", reasons)
```

Comment:
To make this more readable, can you change this to:

```
var reasonStrings []string
for k, v := range reasons {
  reasonStrings = append(reasonStrings, fmt.Sprintf("%s (%d)", k , v))
}
sort.Sort(reasonStrings)
reasonMsg := fmt.Sprintf(..., strings.Join(reasons, ", ")
```

That's easier to read and also makes the result stable (so if we send the same event we don't generate a separate message).